image_transport/include/image_transport/publisher_plugin.hpp

```diff
@@ -63,14 +66,36 @@ class PublisherPlugin
    * \brief Advertise a topic, simple version.
    */
   void advertise(
-    rclcpp::Node * nh,
+    NodeType * nh,
     const std::string & base_topic,
     rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
     rclcpp::PublisherOptions options = rclcpp::PublisherOptions())
   {
-    advertiseImpl(nh, base_topic, custom_qos, options);
+    if (!node_) {
+      node_.reset(nh);
+    }
+    advertise(base_topic, custom_qos, options);
   }
 
+  void advertise(
+    std::shared_ptr<NodeType> nh,
+    const std::string & base_topic,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
+    rclcpp::PublisherOptions options = rclcpp::PublisherOptions())
+  {
+    if (!node_) {
+      node_ = nh;
+    }
+    advertise(base_topic, custom_qos, options);
+  }
+
+  void advertise(
+    const std::string & base_topic,
+    rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
+    rclcpp::PublisherOptions options = rclcpp::PublisherOptions())
+  {
+    advertiseImpl(base_topic, custom_qos, options);
+  }
```

Comment:
what happens if this version of advertise is called but node_ is not set? I'm not sure this version of the function should exist (especially as a public member)

Reply:
Fixed